x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/operator/OperatorPrivileges.java

```diff
@@ -102,7 +102,14 @@ public void maybeMarkOperatorUser(Authentication authentication, ThreadContext t
             final User user = authentication.getEffectiveSubject().getUser();
             // Let internal users pass, they are exempt from marking and checking
             // Also check run_as, it is impossible to run_as internal users, but just to be extra safe
+            // mark internalUser with operator privileges
             if (user instanceof InternalUser && false == authentication.isRunAs()) {
+                if (threadContext.getHeader(AuthenticationField.PRIVILEGE_CATEGORY_KEY) == null) {
+                    threadContext.putHeader(
+                        AuthenticationField.PRIVILEGE_CATEGORY_KEY,
+                        AuthenticationField.PRIVILEGE_CATEGORY_VALUE_OPERATOR
+                    );
+                }
```

Comment:
Discussed with @pgomulka in person. The act of adding this here isn't too much of a concern, all internal users should have full access to roles with out operator privs kicking in and it would be kinda silly to externalize our internal users into a config file.  At a high level the only concern by adding here is that it means it read elsewhere to implement operator privs, but that is already a concern due to the public global nature of thread context... we are just extending that potential to internal users that already exists for external users. 

We discussed a different PR that reads this header value but in a different location than here in this class. That usecase is why this is being added to this class and there are many tradeoffs to consider. The largest tradeoff is that operators privs, for that use case, will not be done via the operator registry. Rather that use case will use action filters to manipulate the responses if users are not operators effectively splitting where operator privs are applied, hence the need to read the thread context in multiple location (and the need to set the context for internal users). Those tradeoffs had already been discussed and accepted (I was not directly part of those). I only have a second hand understanding of those discussions and tradeoffs, but it that seems like an acceptable decision given circumstances and that decision necessitates this change. However, I did encourage that any strategy that implements operator privs outside of the registry have some association back to the registry so there is at least some bread crumbs that allow a dev to know where to look for non-operator related behavior. 

also ++ to remove from the checking side.